src/main/java/com/terraformersmc/terrestria/feature/trees/BryceTreeFeature.java

```diff
@@ -139,28 +134,19 @@ public void placeRoot(Set<BlockPos> blocks, ModifiableTestableWorld world, Block
 			if (isAir(world, pos)) {
 				setBlockStateAndUpdate(blocks, world, pos, this.getLog(), Direction.DOWN, boundingBox);
 			} else {
-				pos.setOffset(Direction.UP);
-				if (((World) world).getBlockState(pos).getBlock() instanceof SmallLogBlock) {
-					setBlockState(world, pos, ((World) world).getBlockState(pos).with(SmallLogBlock.DOWN, true), boundingBox);
-				}
 				break;
 			}
 		}
 	}
 
-	private void placeLeaves(Set<BlockPos> blocks, ModifiableTestableWorld world, BlockBox boundingBox, BranchedTreeFeatureConfig config) {
-		Random random = new Random();
-		BlockPos.Mutable mPos;
-		for (BlockPos pos : leafOrigins) {
-			mPos = new BlockPos.Mutable(pos);
-			for (int i = 0; i < 2; i++) {
-				Shapes.circle(new BlockPos.Mutable(mPos.toImmutable()), 1.0, position -> {
-					tryPlaceLeaves(world, position, random, blocks, boundingBox, config);
-				});
-				mPos.setOffset(Direction.UP);
-			}
-			tryPlaceLeaves(world, mPos, random, blocks, boundingBox, config);
+	private void placeLeaves(ModifiableTestableWorld world, Random random, BlockPos.Mutable pos, Set<BlockPos> blocks, BlockBox boundingBox, TreeFeatureConfig config) {
+		for (int i = 0; i < 2; i++) {
+			Shapes.circle(new BlockPos.Mutable(pos.toImmutable()), 1.0, position -> {
+				tryPlaceLeaves(world, pos, random, blocks, boundingBox, config);
```

Comment:
Not actually drawing a circle here...
```suggestion
				tryPlaceLeaves(world, position, random, blocks, boundingBox, config);
```

Reply:
good catch